packages/backend/src/core/Clock.ts

```diff
@@ -5,25 +5,44 @@ export class Clock {
     private readonly minTimestamp: UnixTime,
     private readonly delayInSeconds: number,
     private readonly refreshIntervalMs = 1000,
-  ) {
-    if (!this.minTimestamp.isFull('hour')) {
-      this.minTimestamp = this.minTimestamp.toNext('hour')
+  ) {}
+
+  getFirstHour(): UnixTime {
+    let result = this.minTimestamp
+
+    if (!result.isFull('hour')) {
+      result = result.toNext('hour')
     }
-    if (this.minTimestamp.gt(this.getLastHour())) {
+    if (result.gt(this.getLastHour())) {
       throw new Error('minTimestamp must be in the past')
     }
+
+    return result
   }
 
-  getFirstHour(): UnixTime {
-    return this.minTimestamp
+  getFirstDay(): UnixTime {
+    let result = this.minTimestamp
+
+    if (!result.isFull('day')) {
+      result = result.toNext('day')
+    }
+    if (result.gt(this.getLastDay())) {
+      throw new Error('minTimestamp must be in the past')
+    }
+
+    return result
   }
 
   getLastHour(): UnixTime {
     return UnixTime.now().add(-this.delayInSeconds, 'seconds').toStartOf('hour')
   }
 
+  getLastDay(): UnixTime {
+    return UnixTime.now().add(-this.delayInSeconds, 'seconds').toStartOf('day')
+  }
+
   onEveryHour(callback: (timestamp: UnixTime) => void) {
-    let next = this.minTimestamp
+    let next = this.getFirstHour()
```

Comment:
A lot of stuff (TVL, Activitity, ?) rely on this loop. How does changing this initial value impact those?